### PR TITLE
[2856] Update the `MessagesScreen` docs

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

Update  `MessagesScreen` docs.

### 🛠 Implementation details

Updated the documentation with new screenshots, code snippets and text where needed.

### 🧪 Testing

Please test the documentation by running Docusaurus and making sure everything renders correctly.

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

message

### CODE REVIEW

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/MessagesActivity.kt [hunk 1]
```diff
@@ -78,7 +78,6 @@ class MessagesActivity : AppCompatActivity() {
             ChatTheme(dateFormatter = ChatApp.dateFormatter) {
                 MessagesScreen(
                     channelId = channelId,
-                    messageLimit = 30,
```
Comment (MarinTolic) [inline stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/MessagesActivity.kt:81]: Not necessary as the default argument for `messageLimit` in `MessagesScreen` is the same

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx [hunk 2]
```diff
@@ -43,7 +44,7 @@ The `MessagesScreenComponent` exposes two actions, as per the signature:
 fun MessagesScreen(
     ..., // State
     onBackPressed: () -> Unit = {},
```
Comment (skydoves) [inline docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:46]: It's not related to this PR, there is a small typo on line 40. `MessagesScreenComponent` should be `MessagesScreen`.

**Review by skydoves — APPROVED** [review]
Looks good to me!

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx [hunk 3]
```diff
@@ -53,42 +54,40 @@ fun MessagesScreen(
...
 
-Given that the `MessagesScreen` is an screen level solution, it offers limited customization. Currently, the only option offered in the signature is the following:
+Given that the `MessagesScreen` is a screen level solution, it offers limited customization. Currently, the only options offered in the signature are the following:
 
 ```kotlin
 @Composable
 fun MessagesScreen(
+    messageLimit: Int = 30,
     showHeader: Boolean = true,
     ... // action handlers and state
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:75]: What about `enforceUniqueReactions`? What do you think, should we mention them?

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx [hunk 4]
```diff
@@ -53,42 +54,40 @@ fun MessagesScreen(
...
-                    },
-                )
-            }
-        }
-    }
+MessagesScreen(
+    ..., // State
+    onBackPressed = { finish() }, // Navigation handler
+    onHeaderActionClick = { channel ->
+        // Show conversation info
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:61]: Hm, maybe we should use the very same terms every time:`Show channel info`

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx [hunk 5]
```diff
@@ -53,42 +54,40 @@ fun MessagesScreen(
...
+    },
+)
```
Comment (bychkovdmitry) [inline docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:63]: Not related to the PR, but maybe we should also invoke the very same action when somebody click on the channel avatar?

**Review by bychkovdmitry — APPROVED** [review]
Great, just couple of small comments

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx (hunk 2 above)
Comment (MarinTolic) [reply docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:46]: Good catch, I'll fix that one :)

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx (hunk 3 above)
Comment (MarinTolic) [reply docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:75]: That is a very valid suggestion and an important part that I had probably overlooked by accident.

I will fix that.

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx (hunk 4 above)
Comment (MarinTolic) [reply docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:61]: Yeah, show channel info is much better and precise than conversation.

File: docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx (hunk 5 above)
Comment (MarinTolic) [reply docusaurus/docs/Android/04-compose/04-message-components/01-messages-screen.mdx:63]: Or maybe expose another Handler for the channel avatar?

**Pull request merged**